packages/datepicker-react/src/DatePicker.tsx

```diff
@@ -112,11 +112,30 @@ export function DatePicker({
         }
     }, [disableBeforeDate, disableAfterDate]);
 
+    const disableDate = useCallback(
+        (date: Date) => {
+            return (disableAfterDate && date > disableAfterDate) || (disableBeforeDate && date < disableBeforeDate);
+        },
+        [disableAfterDate, disableBeforeDate],
+    );
+
     useEffect(() => {
         setDateString(initialDate ? formatDate(initialDate) : "");
         setDate(initialDate);
     }, [initialDate]);
 
+    useEffect(() => {
+        // if initialdate is outside scope, clear it out. this is to avoid a case in wizard forms
+        // where the initialdate may be based on an invalid value earlier in the form state.
+        if (initialDate && disableDate(initialDate)) {
+            setDateString("");
+            setDate(undefined);
+        }
+        // this check should only have effect at first render. this is to ensure a good user experience, where having
+        // it on every change would clear out the field while the user is typing.
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
```

Comment:
burde ikke initalDate være med? Hvis den for some reason skulle endre seg, burde ikke denne effekten fyre igjen?

Reply:
Tanken bak å eksplisitt ha det på initial render, er at det vil lede til ganske kjip oppførsel når man bruker DatePicker som en controlled komponent hvor initialDate endrer seg basert på state. Så om man skriver inn feil dato, vil feltet bare bli tomt med en gang i motsetning til at man får sjansen til å rette opp feilen sin før man submitter.

Reply:
> når man bruker DatePicker som en controlled komponent hvor initialDate endrer seg basert på state

Her høres det vel egentlig ut som man bør ha en annen prop. Det gir i hvert fall ikke mening for meg at man konstant endrer en prop som starter på `initial`. Foreslår å legge til en prop `value` som kan brukes hvis man vil ha en kontrollert komponent, og så heller sette den lik `initialDate` på første render eller noe slikt.

Og oppførselen du beskriver er ganske enkel å unngå ved å slenge inn en test for om verdien er en gyldig dato 😊 Du kan bruke `parseDateString()` fra `dateFunctions.ts` til dette. Den returnerer `undefined` hvis ikke den klarer å parse datoen, så da kan man f.eks. si
```ts
if (parseDateString(value) && disableDate(value)) {
    setDateString("");
    setDate(undefined);
}
```

Reply:
Enig i at å legge inn value kan være lurt, og ikke la initialDate være kontrollerende utover at den er initial.

Ah, clevuhr fix! 😄